viewer/viewer.js

```diff
@@ -7012,180 +7012,173 @@ app.get('/:nodeName/hunt/:huntId/remote/:sessionId', function (req, res) {
+        let opcap = Pcap.get('write'+fields.node + ':' + fileNum);
+        if (!opcap.isOpen()) {
+          Db.fileIdToFile(fields.node, fileNum, function (file) {
+            if (!file) {
+              console.log('WARNING - Only have SPI data, PCAP file no longer available.  Couldn\'t look up in file table', fields.node + '-' + fileNum);
+              return nextCb('Only have SPI data, PCAP file no longer available for ' + fields.node + '-' + fileNum);
+            }
 
-          var ipcap = Pcap.get("write"+fields.node + ":" + file.num);
+            var ipcap = Pcap.get('write'+fields.node + ':' + file.num);
 
-          try {
-            ipcap.openReadWrite(file.name, file);
-          } catch (err) {
-            console.log("ERROR - Couldn't open file for writing", err);
-            return nextCb("Couldn't open file for writing " + err);
-          }
+            try {
+              ipcap.openReadWrite(file.name, file);
+            } catch (err) {
+              console.log('ERROR - Couldn\'t open file for writing', err);
+              return nextCb('Couldn\'t open file for writing ' + err);
+            }
 
-          processFile(ipcap, pos, itemPos++, nextCb);
-        });
-      } else {
-        processFile(opcap, pos, itemPos++, nextCb);
-      }
-    },
-    function (pcapErr, results) {
-      if (entire) {
-        Db.deleteDocument(session._index, 'session', session._id, function(err, data) {
-          endCb(pcapErr, fields);
-        });
-      } else {
-        // Do the ES update
-        var document = {
-          doc: {
-            scrubby: req.user.userId || "-",
-            scrubat: new Date().getTime()
-          }
-        };
-        Db.update(session._index, 'session', session._id, document, function(err, data) {
-          endCb(pcapErr, fields);
-        });
-      }
-    });
+            processFile(ipcap, pos, itemPos++, nextCb);
+          });
+        } else {
+          processFile(opcap, pos, itemPos++, nextCb);
+        }
+      },
+      function (pcapErr, results) {
+        if (whatToRemove === 'all') { // also remove the session data
+          Db.deleteDocument(session._index, 'session', session._id, function (err, data) {
+            return endCb(pcapErr, fields);
+          });
+        } else { // just set who/when scrubbed the pcap
+          // Do the ES update
+          const document = {
+            doc: {
+              scrubby: req.user.userId || '-',
+              scrubat: new Date().getTime()
+            }
+          };
+          Db.update(session._index, 'session', session._id, document, function (err, data) {
+            return endCb(pcapErr, fields);
+          });
+        }
+      });
+    }
   });
 }
 
-app.get('/:nodeName/scrub/:sid', checkProxyRequest, function(req, res) {
-  if (!req.user.removeEnabled) { return res.molochError(200, "Need remove data privileges"); }
+app.get('/:nodeName/delete/:whatToRemove/:sid', checkProxyRequest, function (req, res) {
```

Comment:
i might have used a different endpoint name here, and left delete so that we can interop with old versions. Not sure if this makes sense.